icu4c/source/i18n/unicode/messageformat2_formattable.h

```diff
@@ -64,6 +65,50 @@ namespace message2 {
         virtual ~FormattableObject();
     }; // class FormattableObject
 
+    /**
+     * The `DateInfo` struct represents all the information needed to
+     * format a date with a time zone. It includes an absolute date and a time zone name,
+     * as well as a calendar name. The calendar name is not currently used.
+     *
+     * @internal ICU 76 technology preview
+     * @deprecated This API is for technology preview only.
+     */
+    struct U_I18N_API DateInfo {
+        /**
+         * Milliseconds since Unix epoch
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UDate date;
+        /**
+         * IANA time zone name; "UTC" if UTC; empty string if value is floating
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UnicodeString zoneName;
+        /**
+         * Calendar name. Note: non-Gregorian calendars not yet implemented.
+         * String is empty if calendar not specified. Proleptic Gregorian calendar
+         * is default.
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UnicodeString calendarName;
```

Comment:
I do not understand why you need createTimeZone() here since all the information you need can be derived from zoneName itself. I will prefer we not adding  more function if it is really needed.

Reply:
I included it because to call `DateFormat::format()` to eventually format the date, a `TimeZone` object has to be constructed (to pass to `DateFormat`'s `adoptTimeZone()` or `setTimeZone()`) method. But it doesn't have to be a method of the `DateInfo` struct (changed in [1c68d51](https://github.com/unicode-org/icu/pull/3012/commits/1c68d511902932ccccaef4f5eb75019e9f6d7d80)).